lib/chat/chunked_files_multikey.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Chat.ChunkedFilesMultikey do
+  @moduledoc false
+
+  alias Chat.Db
+  alias Chat.Utils
+
+  @chunk_group_size 1_000_000_000
+  @initial_offset_range 0..1000
+
+  def generate(file_key, file_size, <<initial_secret::binary-size(32)>>) do
+    secrets_needed_amount = trunc(file_size / @chunk_group_size)
+
+    secrets_needed =
+      for amount <- 0..secrets_needed_amount,
+          amount > 0,
+          do: encrypt_secret(additional_secret(), initial_secret)
+
+    Db.put({:file_secrets, file_key}, secrets_needed |> Enum.join(""))
+  end
+
+  def get_secret(_file_key, offset, <<initial_secret::binary-size(32)>>)
+      when offset in @initial_offset_range,
```

Comment:
what is faster ?
 - in range
 - comparison with a number

Reply:
Name                 ips        average  deviation         median         99th %
comparison        1.78 K        0.56 ms     ±8.36%        0.55 ms        0.74 ms
range             0.79 K        1.26 ms     ±8.16%        1.24 ms        1.55 ms

Comparison:
comparison        1.78 K
range             0.79 K - 2.24x slower +0.70 ms

Memory usage statistics:

Name          Memory usage
comparison         0.97 MB
range              2.17 MB - 2.24x memory usage +1.20 MB

```elixir
rand = fn -> :rand.uniform(10*1000*1024*1024) end
list = 
  1..1000
  |> Enum.map(fn _ -> rand.() end)

range_fn = fn n -> n in 0..1000*1024*1024 end
comparison_fn = fn n -> n<=1000*1024*1024 end


res = Benchee.run(
  %{
    "range" => fn -> Enum.map(list, range_fn) end,
    "comparison" => fn -> list |> Enum.map(comparison_fn) end
  },
  time: 10,
  memory_time: 2
)
:ok
```

Reply:
comparison applied